Fix MinGW build failure due to vasprintf

legends2k opened the pull request:
MinGW’s compiler (GCC 14.2.0) errors out compiling `synctex_parser.c`.

```
synctex_parser.c: In function '_synctex_updater_print_gz':
synctex_parser.c:8448:13: error: implicit declaration of function
'vasprintf'; did you mean 'vsprintf'?
  8448 |     if (vasprintf(&buffer, format, va) < 0) {
    |       ^~~~~~~~~
    |       vsprintf
```

`vasprintf` is a custom function (not the standard `vsprintf`) that’s defined in the same translation unit a few lines above.

However its definition is wrapped within a `if defined(_MSC_VER)` i.e. it’s only enabled when compiled with MSVC compiler.  This fix enables it also for MinGW compilers.

Fixes #286.

danielkrizian commented:
i am getting the same error with a compiler different than MinGW, but the https://github.com/vedang/pdf-tools/pull/295 only addresses the MinGW scenario. 
`ifndef HAVE_VASPRINTF` should fix the issue as per 

https://lists.gnu.org/archive/html/emacs-elpa-diffs/2022-08/msg01485.html
https://lists.gnu.org/archive/html/emacs-elpa-diffs/2022-08/msg01533.html

legends2k commented:
The function `vasprintf`, available on [Linux][1] and [macOS][2], is unavailable on Windows irrespective of compiler.

I've updated the PR to address all Windows builds in general; not specifically MinGW or MSVC.  I've changed the implementation to use the standard C library function `vsnprintf` to implement this `vasprintf` which mimics the behaviour of the ones available on Linux/macOS.

[1]: https://linux.die.net/man/3/vasprintf
[2]: https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man3/vasprintf.3.html

chuchana commented:
Thank you @legends2k , that fixed it for me!

legends2k commented:
This is a subset of PR #308.
